Assets/PlayroomKit/PlayroomKit.cs

```diff
@@ -1085,15 +1087,21 @@ private static void InvokeOnResponseCallback()
                             callback?.Invoke();
                         }
 
-                        RpcEventNames.Remove(name);
-                        OnResponseCallbacks.Remove(name);
+                        namesToRemove.Add(name);
                     }
                 }
                 catch (Exception ex)
                 {
                     Debug.LogError($"C#: Error in Invoking callback for RPC event name: '{name}': {ex.Message}");
                 }
             }
+
+            // Remove the names after the loop
+            foreach (var name in namesToRemove)
+            {
+                RpcEventNames.Remove(name);
+                OnResponseCallbacks.Remove(name);
```

Comment:
What if the name does not exist in onResponseCallbacks? Will this crash? Similar question for the line above this.

Reply:
https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.remove?view=net-8.0#:~:text=Returns&text=true%20if%20the%20element%20is,Dictionary.

```
Returns
[Boolean](https://learn.microsoft.com/en-us/dotnet/api/system.boolean?view=net-8.0)
true if the element is successfully found and removed; otherwise, false. This method returns false if key is not found in the [Dictionary<TKey,TValue>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-8.0).
```

If that's the case, then it should work fine.

Reply:
```C#
try
                {
                    if (OnResponseCallbacks.TryGetValue(name, out List<Action> callbacks))
                    {
                        foreach (var callback in callbacks)
                        {
                            callback?.Invoke();
                        }

                        namesToRemove.Add(name);
                    }
                }
```

name is added to this namesToRemove list if it exists in the ```OnResponseCallbacks``` dictionary. So, there shouldn't be any issues.